src/app/app-routing.module.ts

```diff
@@ -0,0 +1,37 @@
+import { NgModule } from '@angular/core';
+import { RouterModule, Routes } from '@angular/router';
+import {AppResolver} from './app-resolver';
+import {NavigationTreeResolver} from './navigation-tree-resolver';
+import {MainWrapperComponent} from './main-wrapper/main-wrapper.component';
+
+const routes: Routes = [
+  { path: '',  redirectTo: '/category?key=1907548051', pathMatch: 'full' },
+  {
+    path: 'category',
+    component: MainWrapperComponent,
+    resolve: {
+      categories: NavigationTreeResolver,
+      products: AppResolver,
+      selectedCategory: AppResolver
```

Comment:
It seems that `selectedCategory` is not used in component. Let's remove this in order not to make duplicated requests.